src/main/java/net/snowflake/ingest/utils/HttpUtil.java

```diff
@@ -135,4 +103,77 @@ public long getRetryInterval()
 
   }
 
+  private static ServiceUnavailableRetryStrategy getServiceUnavailableRetryStrategy()
+  {
+    return new ServiceUnavailableRetryStrategy() {
+      private int executionCount = 0;
+      int REQUEST_TIMEOUT = 408;
+
+      @Override
+      public boolean retryRequest(
+              final HttpResponse response, final int executionCount,
+              final HttpContext context)
+      {
+        this.executionCount = executionCount;
+        int statusCode = response.getStatusLine().getStatusCode();
+        LOGGER.info("In retryRequest for service unavailability with statusCode:{} and uri:{}",
+                    statusCode, getRequestUriFromContext(context));
+        boolean needNextRetry = (statusCode == REQUEST_TIMEOUT || statusCode >= 500)
+                && executionCount < MAX_RETRIES + 1;
+        if (executionCount == MAX_RETRIES + 1)
+        {
+          LOGGER.info("Reach the max retry time.");
+        }
+        if (needNextRetry && executionCount < MAX_RETRIES + 1)
+        {
+          long interval = (1 << executionCount) * 1000;
```

Comment:
Yes I suppose, can remove this though.

Reply:
I don't think it's useful, it prints fixed set of numbers.  What will be useful, is tracking whether or not requests succeed after N retries.  This can be used for tuning the parameters of the policy, and perhaps eliminating it altogether.  Without such tracking  I can't tell whether this entire class adds value or causes more damage.

Reply:
It is going to be hard to track this, I think this is more useful for client logging/debugging.